pkg/controller/rdconfig.go

```diff
@@ -21,11 +21,150 @@ import (
 	"os"
 	"reflect"
 
-	rdConfigv1 "github.com/noironetworks/aci-containers/pkg/rdconfig/apis/aci.snat/v1"
+	rdConfig "github.com/noironetworks/aci-containers/pkg/rdconfig/apis/aci.snat/v1"
+	rdconfigclset "github.com/noironetworks/aci-containers/pkg/rdconfig/clientset/versioned"
+	"github.com/noironetworks/aci-containers/pkg/util"
+	"github.com/sirupsen/logrus"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
 )
 
+func (cont *AciController) initRdConfigInformerFromClient(
+	rdConfigClient *rdconfigclset.Clientset) {
+	ns := os.Getenv("ACI_SNAT_NAMESPACE")
+	name := os.Getenv("ACI_RDCONFIG_NAME")
+	cont.initRdConfigInformerBase(
+		&cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.FieldSelector = fields.Set{"metadata.name": name}.String()
+				obj, err := rdConfigClient.AciV1().RdConfigs(ns).List(context.TODO(), options)
+				if err != nil {
+					cont.log.Fatalf("Failed to list RdConfigs during initialization of RdConfigInformer with err: %v", err)
+				}
+				return obj, err
+			},
+			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+				options.FieldSelector = fields.Set{"metadata.name": name}.String()
+				obj, err := rdConfigClient.AciV1().RdConfigs(ns).Watch(context.TODO(), options)
+				if err != nil {
+					cont.log.Fatalf("Failed to watch RdConfigs during initialization RdConfigsInformer with err: %v", err)
+				}
+				return obj, err
+			},
+		})
+}
+
+func (cont *AciController) initRdConfigInformerBase(listWatch *cache.ListWatch) {
+	cont.rdConfigIndexer, cont.rdConfigInformer = cache.NewIndexerInformer(
+		listWatch,
+		&rdConfig.RdConfig{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.RdConfigAdded(obj)
+			},
+			UpdateFunc: func(oldobj interface{}, newobj interface{}) {
+				cont.RdConfigUpdated(oldobj, newobj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.RdConfigDeleted(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing RdConfig Informers")
+}
+
+func (cont *AciController) RdConfigAdded(obj interface{}) {
```

Comment:
is it a void function return? Do we not have to return err/pass

Reply:
Yes, added, updated and deleted are void functions. They plug in to the Resource Handler Interface through the `initRdConfigInformerBase` function above. This interface does not define a return type for the functions.